Add docs for Music Assistant integration (#35494)
home-assistant/home-assistant.github.io: source/_integrations/music_assistant.markdown

Play media hosted on a Music Assistant server on a Music Assistant player. The action configuration is as described in the [media player documentation](https://www.home-assistant.io/integrations/media_player/#action-media_playerplay_media)

The `media_content_id` payload can be any of the following:

- The name of a track, artist or album. (for example, "Queen")
- A track or album combined with the artist name (for example, "Queen - Innuendo")
- A streaming provider URI (for example, `spotify://artist/12345`)

#### Examples:

Play Adele's album 25

```yaml
entity_id: media_player.music_assistant_player
media_content_type: MUSIC
media_content_id: 'Adele - 25'
```

Play all tracks from Stevie Wonder

```yaml
entity_id: media_player.music_assistant_player
media_content_type: MUSIC
media_content_id: 'Stevie Wonder'
```

Play the playlist The Best of Disco

```yaml
entity_id: media_player.music_assistant_player
media_content_type: PLAYLIST
media_content_id: 'The Best of Disco'
```

## Additional actions

### Action `music_assistant.play_media`

Play media on a Music Assistant player with more fine-grained control options.

- **Data attribute**: `media_id`
  - **Optional**: No.

>   - **Description**: URI or name of the item to be play. Specify a list if it is desired to play/enqueue multiple items.

Reviewer: ```suggestion
  - **Description**: URI or name of the item to be played. Specify a list if it is desired to play/enqueue multiple items.
```

    - play: Play now
    - replace: Replace the existing queue and play now
    - next: Add to the current queue after the currently playing item
    - replace_next: Replace the current queue after the currently playing item
    - add: Add to the end of the queue
  - **Example**: `replace`
- **Data attribute**: `radio_mode`
  - **Optional**: Yes.
  - **Description**: Enable radio mode to auto-generate a playlist based on the selection.
  - **Example**: `true`

### Action `music_assistant.play_announcement`

Play announcement on a Music Assistant player with more fine-grained control options.

- **Data attribute**: `url`
  - **Optional**: No.
  - **Description**: URL to the notification sound.
  - **Example**: `https://someremotesite.com/doorbell.mp3`
- **Data attribute**: `use_pre_announce`
  - **Optional**: Yes.
  - **Description**: Use pre-announcement sound. Omit to use the player default.
  - **Example**: `true`
- **Data attribute**: `announce_volume`
  - **Optional**: Yes.
  - **Description**: Use a forced volume level for the announcement. Omit to use the player default.
  - **Example**: `75`

> ### Action `music_assistant.search`
> 
> Perform a global search on the Music Assistant library and all providers.
> 
> - **Data attribute**: `name`
>   - **Optional**: No.
>   - **Description**: The name/title to search for.
>   - **Example**: `We Are The Champions`
> - **Data attribute**: `media_type`
>   - **Optional**: Yes.
>   - **Description**: The type(s) of content to search for. Select from artist, album, track, radio, or playlist. All types if omitted.
>   - **Example**: `playlist`
> - **Data attribute**: `artist`
>   - **Optional**: Yes.
>   - **Description**: When specifying a track or album in the name field, you can optionally restrict results by this artist name.
>   - **Example**: `Queen`
> - **Data attribute**: `album`
>   - **Optional**: Yes.
>   - **Description**: When specifying a track in the name field, you can optionally restrict results by this album name.
>   - **Example**: `News of the world`
> - **Data attribute**: `limit`
>   - **Optional**: Yes.
>   - **Description**: Maximum number of items to return (per media type).
>   - **Example**: `10`
> - **Data attribute**: `library_only`
>   - **Optional**: Yes.
>   - **Description**: Only include results that are in the library.
>   - **Example**: `true`

Reviewer 2: ```suggestion
```

search action was (last minute) removed from the PR, awaiting further discussion

> 
> ### Action `music_assistant.get_library`
> 
> Perform a local search on the Music Assistant library.
> 
> - **Data attribute**: `media_type`
>   - **Optional**: No.
>   - **Description**: The type of content to search for. Select from artist, album, track, radio, or playlist.
>   - **Example**: `artist`
> - **Data attribute**: `favorite`
>   - **Optional**: Yes.
>   - **Description**: When selected only items marked as favorites will be returned.
>   - **Example**: `false`
> - **Data attribute**: `limit`
>   - **Optional**: Yes.
>   - **Description**: Maximum number of items to return.
>   - **Example**: `25`
> - **Data attribute**: `offset`
>   - **Optional**: Yes.
>   - **Description**: From what point in the list should results be returned.
>   - **Example**: `10`
> - **Data attribute**: `search`
>   - **Optional**: Yes.
>   - **Description**: A string that will further filter the results.
>   - **Example**: `Home`
> - **Data attribute**: `order_by`
>   - **Optional**: Yes.
>   - **Description**: Sort the list by this field. The options are numerous so inspect them in the Developer Tools, Actions tab.
>   - **Example**: `year`
> - **Data attribute**: `album_artists_only`
>   - **Optional**: Yes.
>   - **Description**: When `artist` is the `media_type` then this option will restrict the result to album artists only.
>   - **Example**: `true`
> - **Data attribute**: `album_type`
>   - **Optional**: Yes.
>   - **Description**: When `album` is the `media_type` then this option will restrict the result according to the selection of either album, single, compilation, EP or unknown.
>   - **Example**: `album`
> 

Reviewer 2: ```suggestion
```

get_library action was (last minute) removed from the PR, awaiting further discussion

> ### Action `music_assistant.get_queue`
> 
> Get the full queue details of a Music Assistant Queue.
> 
> - **Data attribute**: `entity_id`
>   - **Optional**: No.
>   - **Description**: The entity_id of the queue to be retrieved.
>   - **Example**: `media_player.kitchen_speaker`
> 
> ### Action `music_assistant.transfer_queue`
> 
> Transfer the player's queue to another player.
> 
> - **Data attribute**: `source_player`
>   - **Optional**: Yes.
>   - **Description**: The source media player which has the queue to be transferred. When omitted, the first playing player will be used.
>   - **Example**: `media_player.kitchen_speaker`
> - **Data attribute**: `auto_play`
>   - **Optional**: Yes.
>   - **Description**: Start playing the queue on the target player. Omit to use the default behavior.
>   - **Example**: `true`
> 

Reviewer 2: ```suggestion
```

get_queue action was (last minute) removed from the PR, awaiting further discussion

The Music Assistant media player creates media player entities for all players available in MA including those imported from Home Assistant. This is needed to provide the full functionality Music Assistant has to offer. These entities will display media information, playback progress, and playback controls.

### Action `media_player.play_media`

Play media hosted on a Music Assistant server on a Music Assistant player. The action configuration is as described in the [media player documentation](https://www.home-assistant.io/integrations/media_player/#action-media_playerplay_media)

The `media_content_id` payload can be any of the following:

> - The name of a track, artist or album. (e.g. "Queen")
> - A track or album combined with the artist name (e.g. "Queen - Innuendo")
> - A streaming provider URI (e.g. `spotify://artist/12345`)

Reviewer: ```suggestion
- The name of a track, artist or album. (for example, "Queen")
- A track or album combined with the artist name (for example, "Queen - Innuendo")
- A streaming provider URI (for example, `spotify://artist/12345`)
```

#### Examples:

Play Adele's album 25

```yaml
entity_id: media_player.music_assistant_player
media_content_type: MUSIC
media_content_id: 'Adele - 25'
```

Play all tracks from Stevie Wonder

```yaml
entity_id: media_player.music_assistant_player
media_content_type: MUSIC
media_content_id: 'Stevie Wonder'
```

Play the playlist The Best of Disco

```yaml
entity_id: media_player.music_assistant_player
media_content_type: PLAYLIST
media_content_id: 'The Best of Disco'
```

## Additional actions

### Action `music_assistant.play_media`

Play media on a Music Assistant player with more fine-grained control options.

| Data attribute | Required | Description                                                                                                              | Example                           |

> | -------------- | -------- | ------------------------------------------------------------------------------------------------------------------------ | --------------------------------- |

Reviewer: Nice docs!
One question: we are leaning towards replacing wide tables by lists, because they work better on mobile. Would you consider converting those tables here?

Author: No obvious reason why not. Do you have an example in the existing docs?

Author: Here maybe? https://www.home-assistant.io/integrations/plex/#music

Reviewer: > No obvious reason why not. Do you have an example in the existing docs?

ofc, sorry.
<https://www.home-assistant.io/integrations/xiaomi_miio/> has a few.
This used to be a table:
![image](https://github.com/user-attachments/assets/3894bd6f-248a-4066-b8d8-eea997ef7a03)

Or this, using a slightly different style
![image](https://github.com/user-attachments/assets/404ba5e3-760e-4f62-af73-035670963346)

Reviewer: > Here maybe? https://www.home-assistant.io/integrations/plex/#music

Ah, yes. They have some examples as well.

Author: I'll change it to the vacuum_goto style. Standby!

Author: Changes made

| `enqueue`      | No       | If the content should be played now or be added to the queue. Options are: play, replace, next, replace_next, add        | "replace"                         |
| `radio_mode`   | No       | Enable radio mode to auto-generate a playlist based on the selection.                                                    | "true"                            |

### Action `music_assistant.play_announcement`

Play announcement on a Music Assistant player with more fine-grained control options.

| Data attribute         | Required | Description                                                                      | Example                                  |
| ---------------------- | -------- | -------------------------------------------------------------------------------- | ---------------------------------------- |
| `url`                  | Yes      | URL to the notification sound.                                                   | "http://someremotesite.com/doorbell.mp3" |
| `use_pre_announce`     | No       | Use pre-announcement sound for the announcement. Omit to use the player default. | "true"                                   |

> | `announce_volume`      | No       | Use a forced volume level for the announcement. Omit to use player default.      | "75"                                     |

Reviewer: ```suggestion
| `announce_volume`      | No       | Use a forced volume level for the announcement. Omit to use the player default.      | "75"                                     |
```

### Action `music_assistant.search`

Perform a global search on the Music Assistant library and all providers.

| Data attribute | Required | Description                                                                                                       | Example                |
| ---------------| -------- | ----------------------------------------------------------------------------------------------------------------- | ---------------------- |
| `name`         | Yes      | The name/title to search for.                                                                                     | "We Are The Champions" |

> | `media_type`   | No       | The type of the content to search. Such as artist, album, track, radio or playlist. All types if omitted.         | "playlist"             |

Reviewer: ```suggestion
| `media_type`   | No       | The type of the content to search. Such as artist, album, track, radio, or playlist. All types if omitted.         | "playlist"             |
```
